domain/consensus/processes/syncmanager/antipast.go

```diff
@@ -99,7 +99,79 @@ func (sm *syncManager) antiPastHashesBetween(lowHash, highHash *externalapi.Doma
 }
 
 func (sm *syncManager) missingBlockBodyHashes(highHash *externalapi.DomainHash) ([]*externalapi.DomainHash, error) {
-	panic("implement me")
+	headerTipsPruningPoint, err := sm.consensusStateManager.HeaderTipsPruningPoint()
+	if err != nil {
+		return nil, err
+	}
+
+	selectedChildIterator, err := sm.dagTraversalManager.SelectedChildIterator(highHash, headerTipsPruningPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	lowHash := headerTipsPruningPoint
+	for selectedChildIterator.Next() {
+		lowHash = selectedChildIterator.Get()
+	}
+
+	hashesBetween, err := sm.antiPastHashesBetween(lowHash, highHash)
+	if err != nil {
+		return nil, err
+	}
+
+	lowHashAnticone, err := sm.dagTraversalManager.AnticoneFromContext(highHash, lowHash)
+	if err != nil {
+		return nil, err
+	}
+
+	blockToRemoveFromHashesBetween := hashset.New()
+	for _, blockHash := range lowHashAnticone {
+		isHeaderOnlyBlock, err := sm.isHeaderOnlyBlock(blockHash)
+		if err != nil {
+			return nil, err
+		}
+
+		if !isHeaderOnlyBlock {
+			blockToRemoveFromHashesBetween.Add(blockHash)
+		}
+	}
+
+	missingBlocks := make([]*externalapi.DomainHash, 0, len(hashesBetween)-len(lowHashAnticone))
+	for i, blockHash := range hashesBetween {
```

Comment:
This loop is quite complicated. Consider simplifying it. In either case, people add a comment that explains it.